### DISCUSSION TITLE
Question: Custom scale function

### DISCUSSION THREAD

**Discussion by parthmehta** [d3/d3#3610] [Q&A]
I have to display Weibull distribution plot as shown in the sample figure. 

For Weibull distribution , formula for Y-Axis Scale is as:

**_ln(-ln(1-p))_**

where p=(i-0.3)/(n+0.4) and i is the rank of the observation. This scale is chosen in order to linearize the resulting plot for Weibull data. You can have a look at the Y Axis log scale as shown in the example graph below.

I tried standard scaleLog() but I need to customize the log function.

`const y_scale = scaleLog().range([0, 100]);`

Please guide for a solution. 

![Fit_Weibull_2P_](https://user-images.githubusercontent.com/2552090/79609089-c2c0d300-8113-11ea-9b94-a63a80ce7df4.png)

**Comment by vvianyy** [comment]
It's Dec. 2022 now, how do you solve this problem？

**Reply by Fil** [reply]
Can you share the dataset? I'd be happy to help set up this custom scale.

**Reply by curran** [reply]
Also, `scaleLog` doesn't do well with a range that starts with `0`. Have you tried https://github.com/d3/d3-scale#scaleSymlog ?

Maybe you could start by making a copy of https://github.com/d3/d3-scale/blob/main/src/log.js and modifying it?

**Reply by vvianyy** [reply]
May a have a guide about how to customize my own scale function with continues input and continues output? The scale function is really complex but it looks like a linear scale before 1000,  and a log scale after 1000.

![1670811073670](https://user-images.githubusercontent.com/2127129/206946110-961be5cb-916d-4ebd-9478-6000257c9e47.png)

It is called a  biexpontial scale (whose shape is similar to an [arcsinh scale](https://www.omiq.ai/guides/understanding-data-scaling) ) for a flow cytrometry dataset. 

Since the dataset could be large( sometimes more than 10G), the plot picture would be calculated in the cloud, the only two things to do with d3 are implementing the x-axis and y-aixs which fit the plot picture.

We are using  axis tick positions and  axis tick maker numbers calculated out in the cloud, what's more, we would also like to see wether we could implement our own scale functions in the browser, in order to support some user interacts such as drawing a gate.

**Reply by vvianyy** [reply]
Here is an example of a gate in a pseudocolor plot(searched in google), if the customized axis scale is available, we could preview them in the browser easier.

![image](https://user-images.githubusercontent.com/2127129/206947006-a625f490-6ad7-4ff6-ae0a-0f4a6f8a18cb.png)

**Reply by Fil** [reply]
I'm happy to help but please don't let me do all the homework. Can you share a reference to the maths? An example of parameters for the function f-1? Standard techniques for computing f? An example dataset to validate expectations? Thanks
